src/inventory_ui.cpp

```diff
@@ -838,70 +838,70 @@ constexpr char first_invlet = '0';
 constexpr char last_invlet = '9';
 typedef std::vector< std::list<item> > pseudo_inventory;
 
-template<typename Collection, typename Filter>
-void pseudo_inv_to_slice( Collection here, Filter filter,
+template<typename Collection, typename Location>
+void pseudo_inv_to_slice( Collection here, Location pos, item_filter filter,
                           pseudo_inventory &item_stacks, indexed_invslice &result_slice,
-                          std::vector<item *> &selectables, char &cur_invlet )
+                          std::vector<std::pair<Location, item *>> &selectables, char &cur_invlet )
 {
+    for (auto& candidate : here) {
+        if (filter(candidate)) {
+            auto stack = std::find_if(item_stacks.begin(), item_stacks.end(),
+                                      [&](const std::list<item>& e) { return candidate.stacks_with(e.back()); });
```

Comment:
Lambdas generally look the most readable when they are written like functions, ie. first line for the declaration - `[]() {` - then the body, then on the last line the closing brace.